src/components/App.tsx

```diff
@@ -70,26 +74,37 @@ class App extends TsxComponent<AppProps> {
     return this.initialize();
   }
 
+  @ProvideReactive(FSXA_INJECT_KEY_TPP_VERSION)
+  get tppVersion() {
+    return this.fsTppVersion || DEFAULT_TPP_SNAP_VERSION;
+  }
+
   mounted() {
     if (this.appState === FSXAAppState.not_initialized) this.initialize();
-
     // we will load tpp-snap, if we are in devMode
     if (this.isEditMode) {
-      // eslint-disable-next-line
-      const TPP_SNAP = require("fs-tpp-api/snap");
-      TPP_SNAP.onRequestPreviewElement((previewId: string) => {
-        const pageId = previewId.split(".")[0];
-        const nextPage = this.navigationData?.idMap[pageId];
-        if (nextPage) this.requestRouteChange(nextPage.seoRoute);
-      });
-      TPP_SNAP.onRerenderView(() => {
-        window.setTimeout(() => this.initialize(), 300);
-        return false;
-      });
-      TPP_SNAP.onNavigationChange(() => {
-        window.setTimeout(() => this.initialize(), 300);
-        return false;
-      });
+      importTPPSnapAPI(this.tppVersion)
+        .then(() => {
+          const TPP_SNAP = getTPPSnap();
```

Comment:
getTPPSnap() may return `null`, so we should check for that in order to prevent errors